pkg/k8shandler/forwarding.go

```diff
@@ -48,7 +49,26 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
 
 	clusterRequest.ForwardingSpec = clusterRequest.normalizeLogForwarding(clusterRequest.cluster.Namespace, clusterRequest.cluster)
 	generator, err := forwarding.NewConfigGenerator(clusterRequest.cluster.Spec.Collection.Logs.Type, clusterRequest.includeLegacyForwardConfig(), clusterRequest.includeLegacySyslogConfig())
-	return generator.Generate(&clusterRequest.ForwardingSpec)
+	generatedConfig, err := generator.Generate(&clusterRequest.ForwardingSpec)
+
+	if err != nil {
+		return "",
+			clusterRequest.UpdateCondition(
+				logging.CollectorDeadEnd,
+				"Collectors are defined but there is no defined LogStore of LogForward destinations",
```

Comment:
Isn't this condition specifically related the our managed logStore not being defined?  This would mean this is misleading in any other context as generally for LogForwarding, we  have no idea if the destination is a dead end.  IMO, we should explicitly reference it is CLO managed logstore and maybe rename the ClusteConditionType accordingly

Reply:
to get an error here we have neither a LogForward destination defined nor a CLO logstore...